contrib/IECoreAlembic/src/IECoreAlembic/CurvesAlgo.cpp

```diff
@@ -0,0 +1,118 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECore/MessageHandler.h"
+
+#include "IECoreAlembic/ObjectAlgo.h"
+#include "IECoreAlembic/CurvesAlgo.h"
+#include "IECoreAlembic/GeomBaseAlgo.h"
+
+using namespace IECore;
+using namespace IECoreAlembic;
+using namespace Alembic::AbcGeom;
+
+namespace IECoreAlembic
+{
+
+namespace CurvesAlgo
+{
+
+IECOREALEMBIC_API IECore::CurvesPrimitivePtr convert( const Alembic::AbcGeom::ICurves &curves, const Alembic::Abc::ISampleSelector &sampleSelector )
+{
+	const ICurvesSchema curvesSchema = curves.getSchema();
+	const ICurvesSchema::Sample sample = curvesSchema.getValue( sampleSelector );
+
+	IntVectorDataPtr vertsPerCurve = new IntVectorData;
+	vertsPerCurve->writable().insert(
+		vertsPerCurve->writable().end(),
+		sample.getCurvesNumVertices()->get(),
+		sample.getCurvesNumVertices()->get() + sample.getCurvesNumVertices()->size()
+	);
+
+	CubicBasisf basis = CubicBasisf::linear();
```

Comment:
There's no default constructor, so I need to do something. Do you have another preference?